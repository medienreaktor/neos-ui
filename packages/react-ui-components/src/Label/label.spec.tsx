import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';

import Label, {LabelProps} from './label';

describe('<Label/>', () => {
    const props: LabelProps = {
        htmlFor: 'test for',
        theme: {
            label: 'labelClassName'
        }

    };

    it('should render correctly.', () => {
        const wrapper = shallow(<Label {...props}/>); // eslint-disable-line jsx-a11y/label-has-for

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should allow the propagation of "className" with the "className" prop.', () => {
        const wrapper = shallow(<Label {...props} className="fooClassName"/>); // eslint-disable-line jsx-a11y/label-has-for

        expect(wrapper.prop('className')).toContain('fooClassName');
    });

    it('should allow the propagation of additional props to the wrapper.', () => {
        const wrapper = shallow(<Label {...props} title="foo"/>); // eslint-disable-line jsx-a11y/label-has-for

        expect(wrapper.prop('title')).toBe('foo');
    });
});
