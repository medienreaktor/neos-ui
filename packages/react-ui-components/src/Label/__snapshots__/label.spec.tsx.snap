// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label/> should render correctly. 1`] = `
<label
  className="labelClassName"
  htmlFor="test for"
/>
`;
